src/common/connection.js

```diff
@@ -176,7 +176,8 @@ class Connection extends EventEmitter {
         this._ws.on('error', error =>
           this.emit('error', 'websocket', error.messsage, error));
         this._ws.on('message', this._onMessage.bind(this));
-        this._onUnexpectedCloseBound = this._onUnexpectedClose.bind(this);
+        this._onUnexpectedCloseBound = this._onUnexpectedClose.bind(this,
+          resolve, reject);
```

Comment:
Yep, you right, missed other case. Problem was when `close` event comes before `open` - that leaves first promise, returned from `connect` unfilled. 
I fixed that now.